packages/core/src/physics/shape/ColliderShape.ts

```diff
@@ -0,0 +1,96 @@
+import { IColliderShape } from "@oasis-engine/design";
+import { PhysicsMaterial } from "../PhysicsMaterial";
+import { Quaternion, Vector3 } from "@oasis-engine/math";
+import { ignoreClone } from "../../clone/CloneManager";
+import { Collider } from "../Collider";
+
+/**
+ * Abstract class for collider shapes.
+ */
+export abstract class ColliderShape {
+  static idGenerator: number = 0;
+
+  /** @internal */
+  _collider: Collider;
+  /** @internal */
+  @ignoreClone
+  _index: number = -1;
+  /** @internal */
+  _nativeShape: IColliderShape;
+
+  protected _id: number;
+  protected _position: Vector3 = new Vector3();
+  protected _rotation: Quaternion = new Quaternion();
+  protected _material: PhysicsMaterial;
+
+  /**
+   * collider owner of this shape.
+   */
+  get collider(): Collider {
+    return this._collider;
+  }
+
+  /**
+   * unique id for this shape.
+   */
+  get id(): number {
+    return this._id;
+  }
+
+  /**
+   * physical material.
```

Comment:
Capitalize the first letter,and the other comments are the same.

Reply:
done